inc/classes/subscriber/Media/class-webp-subscriber.php

```diff
@@ -0,0 +1,593 @@
+<?php
+namespace WP_Rocket\Subscriber\Media;
+
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Subscriber\CDN\CDNSubscriber;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+use WP_Rocket\Subscriber\Third_Party\Plugins\Images\Webp\Webp_Interface;
+
+/**
+ * Subscriber for the WebP support.
+ *
+ * @since  3.4
+ * @author Remy Perona
+ * @author Grégory Viguier
+ */
+class Webp_Subscriber implements Subscriber_Interface {
+	/**
+	 * Options instance.
+	 *
+	 * @var    Options_Data
+	 * @access private
+	 * @author Remy Perona
+	 */
+	private $options;
+
+	/**
+	 * CDNSubscriber instance.
+	 *
+	 * @var    CDNSubscriber
+	 * @access private
+	 * @author Grégory Viguier
+	 */
+	private $cdn_subscriber;
+
+	/**
+	 * \WP_Filesystem_Direct instance.
+	 *
+	 * @var    \WP_Filesystem_Direct
+	 * @access private
+	 * @author Grégory Viguier
+	 */
+	private $filesystem;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since  3.4
+	 * @access public
+	 * @author Remy Perona
+	 *
+	 * @param Options_Data  $options       Options instance.
+	 * @param CDNSubscriber $cdn_subsciber CDNSubscriber instance.
+	 */
+	public function __construct( Options_Data $options, CDNSubscriber $cdn_subsciber ) {
+		$this->options       = $options;
+		$this->cdn_subsciber = $cdn_subsciber;
+	}
+
+	/**
+	 * Returns an array of events that this subscriber wants to listen to.
+	 *
+	 * @since  3.4
+	 * @access public
+	 * @author Remy Perona
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'rocket_buffer'                   => [ 'convert_to_webp', 23 ],
+			'rocket_webp_section_description' => 'webp_section_description',
+			'rocket_disable_webp_cache'       => 'maybe_disable_webp_cache',
+			'rocket_third_party_webp_change'  => 'sync_webp_cache_with_third_party_plugins',
+		];
+	}
+
+	/** ----------------------------------------------------------------------------------------- */
+	/** HOOKS =================================================================================== */
+	/** ----------------------------------------------------------------------------------------- */
+
+	/**
+	 * Converts images extension to WebP if the file exists.
+	 *
+	 * @since  3.4
+	 * @access public
+	 * @author Remy Perona
+	 * @author Grégory Viguier
+	 *
+	 * @param string $html HTML content.
+	 * @return string
+	 */
+	public function convert_to_webp( $html ) {
+		if ( ! $this->options->get( 'cache_webp' ) || $this->get_plugins_serving_webp() ) {
```

Comment:
Shouldn't it be `! $this->get_plugins_serving_webp()`?

Reply:
No.
If a plugin is serving webp, then we don't need to do anything.